### PR TITLE
Issue #334

### PR DESCRIPTION
**Pull request by kirjaamo**
Progress reporting documented.

### CODE REVIEW

**Comment by kirjaamo** [comment]
Or do these reporting methods belong to the sender's calling conventions?

**Review by byte — COMMENTED** [review]
for scriptable-sst.rst, i am not sure if this is actually doable. can you ask alexey?

for sst-physical.rst, this is ok, but have you noticed on line 130 it is spelt CLuster instead of Cluster? Please fix the capital L too

File: galeracluster/source/documentation/scriptable-sst.rst [hunk 1]
```diff
@@ -103,6 +103,12 @@ These parameters are passed to all state transfer scripts, regardless of method
 
    ``--defaults-file`` The script is given the path to the ``my.cnf`` configuration file.
 
+   ``--total N`` This progress reporting parameter indicates the new SST stage and reports the estimated total work.
+
+   ``--complete N`` This progress reporting parameter reports the work completed so far.
+   
+   Progress reporting is also enabled for the ``clone`` SST method.
```
Comment (ayurchen) [inline galeracluster/source/documentation/scriptable-sst.rst:110]: This is a correct statement, but I'm not sure it belongs in the list of SST script parameters.

File: galeracluster/source/documentation/scriptable-sst.rst [hunk 2]
```diff
@@ -103,6 +103,12 @@ These parameters are passed to all state transfer scripts, regardless of method
 
    ``--defaults-file`` The script is given the path to the ``my.cnf`` configuration file.
 
+   ``--total N`` This progress reporting parameter indicates the new SST stage and reports the estimated total work.
```
Comment (ayurchen) [inline galeracluster/source/documentation/scriptable-sst.rst:106]: This is not an SST script parameter, just like "complete". This is optional script output. For example, the script output on DONOR over the period of SST may look like:
```
flush tables
total 534
continue
complete 122
complete 340
complete 610
total 23
complete 20
done
```
Here we had two phases, one estimated 543 units of work length, another 23. First phase estimation was too small and we ended up doing a bit more. Second phase estimation may have been correct, but it was over before the full progress could be reported. Both `total` and `complete` messages are optional. In  the absence of those, we conclude that progress cannot be evaluated and just wait for `done` that unequivocally indicates that SST is finished.

The script outputs those control messages to standard output from where they are read by the parent mysqld process.

File: galeracluster/source/documentation/scriptable-sst.rst [hunk 3]
```diff
@@ -103,6 +103,12 @@ These parameters are passed to all state transfer scripts, regardless of method
...
+
+   ``--complete N`` This progress reporting parameter reports the work completed so far.
```
Comment (ayurchen) [inline galeracluster/source/documentation/scriptable-sst.rst:108]: Same as above. These are not script parameters, these are script feedback to the caller.

**Review by ayurchen — CHANGES_REQUESTED** [review]
Changes are needed, see inline comments

**Comment by kirjaamo** [comment]
@byte I could not resolve one conflict, so I created a new branch for it and made a pull request a minute ago.

Your comments for _scriptable-sst.rst_ are implemented in the original branch.

I hope this hassle does not complicate your life too much

**Review by byte — APPROVED** [review]
ok

**Pull request merged**